EngineLayer/Localization/LocalizationEngine.cs

```diff
@@ -51,6 +52,12 @@ protected override MetaMorpheusEngineResults RunSpecific()
                 }
 
                 ok.MatchedIonDictPositiveIsMatch = new MatchedIonMassesListPositiveIsMatch(matchedIonDictPositiveIsMatch);
```

Comment:
Let's change this so that only the matched ions are present in the  ok.MatchedIonMassesListPositiveIsMatch, and not the unmatched ones. This needs to be done, because in case of ambiguous compact peptides, the unmatched peaks might be different. It wasn't a problem before, since the computation was only done on unambiguous compact peptides.